src/apprt/gtk-ng/class/command_palette.zig

```diff
@@ -0,0 +1,426 @@
+    pub inline fn isSupported() bool {
+        return comptime adw_version.supportsDialogs();
+    }
+
+    const Self = @This();
+    parent_instance: Parent,
+    pub const Parent = adw.Bin;
+    pub const getGObjectType = gobject.ext.defineClass(Self, .{
+        .name = "GhosttyCommandPalette",
+        .instanceInit = &init,
+        .classInit = &Class.init,
+        .parent_class = &Class.parent,
+        .private = .{ .Type = Private, .offset = &Private.offset },
+    });
+
+    pub const properties = struct {
+        pub const config = struct {
+            pub const name = "config";
+            const impl = gobject.ext.defineProperty(
+                name,
+                Self,
+                ?*Config,
+                .{
+                    .nick = "Config",
+                    .blurb = "The configuration that this command palette is using.",
+                    .accessor = gobject.ext.privateFieldAccessor(
+                        Self,
+                        Private,
+                        &Private.offset,
+                        "config",
+                    ),
+                },
+            );
+        };
+    };
+
+    const Private = struct {
+        /// The configuration that this command palette is using.
+        config: ?*Config = null,
+
+        /// The window that the command palette is attached to.
+        window: ?*Window = null,
+
+        arena: ArenaAllocator,
+
+        /// The dialog object containing the palette UI.
+        dialog: *adw.Dialog,
+
+        /// The search input text field.
+        search: *gtk.SearchEntry,
+
+        /// The view containing each result row.
+        view: *gtk.ListView,
+
+        /// The model that provides filtered data for the view to display.
+        model: *gtk.SingleSelection,
+
+        /// The list that serves as the data source of the model.
+        /// This is where all command data is ultimately stored.
+        source: *gio.ListStore,
+
+        pub var offset: c_int = 0;
+    };
+
+    /// Create a new instance of the command palette, connected to the given
+    /// window, and using the given config. The config should already have had
+    /// it's reference count increased and we take ownership of it. The instance
+    /// returned will be floating.
+    pub fn new(window: *Window, config: *Config) ?*Self {
+        if (comptime !Self.isSupported()) return null;
+
+        const self = gobject.ext.newInstance(Self, .{});
+
+        const app = Application.default();
+        const priv = self.private();
+
+        // The config that we will use to begin with. It should already have
+        // had it's reference count increased, and we take ownership of it.
+        priv.config = config;
+
+        // Note that we _do not_ take a reference to the window as this can
+        // cause a circular reference count, which means that the window/command
+        // palette pair will never be destroyed`.
+        priv.window = window;
```

Comment:
I'm curious to look into this... feels like during `dispose` of `Window` we should release the reference?

Reply:
I tried that, but then Ghostty wouldn't quit - the dispose virtual calls were never called because there was always at least one reference.

Reply:
Yeah a WeakRef seems to work well.